### PR TITLE
Modification in `MarginalDistribution.pdf`

### PR DESCRIPTION
**Pull request by akash9712**
Added check in `MarginalDistribution.pdf` to check for iterables in joint_rv.py. Added test.

#### Brief description of what is fixed or changed
Changes were made as per the review comment [here](https://github.com/sympy/sympy/pull/14989#discussion_r206074904)

#### Release Notes

NO ENTRY

### CODE REVIEW

**Comment by akash9712** [comment]
Ping @Upabjojr

File: sympy/stats/joint_rv.py [hunk 1]
```diff
@@ -284,6 +284,8 @@ def symbols(self):
         return set([rv.pspace.symbol for rv in rvs])
 
     def pdf(self, x, evaluate=False):
+        if not hasattr(x, '__iter__'):
```
Comment (Upabjojr) [inline sympy/stats/joint_rv.py]: why this `hasattr`?

Comment (Upabjojr) [reply sympy/stats/joint_rv.py]: Maybe the joint pdf should simply be defined with `pdf(self, *x)`.

Comment (akash9712) [reply sympy/stats/joint_rv.py]: Should I remove the evaluate flag, then?

Comment (Upabjojr) [reply sympy/stats/joint_rv.py]: I would say that it's best to keep the API compatible with the rest of the module. So yes, just *x

**Comment by akash9712** [comment]
Ping @Upabjojr

**Pull request merged**